Project/Src/Gui/TextEditorControlBase.cs

```diff
@@ -36,6 +36,7 @@ public abstract class TextEditorControlBase : UserControl
         ///     action.
         /// </summary>
         protected Dictionary<Keys, IEditAction> editactions = new Dictionary<Keys, IEditAction>();
+        protected HashSet<Keys> readonlyactions = new HashSet<Keys>();
```

Comment:
```suggestion
        protected HashSet<Keys> readonlyonlyactions = new HashSet<Keys>();
```
or add a regarding xmldoc. Without a clarification here or in `IsEditAction` it is hard to grasp.